synchronization/trigger_test.go

```diff
@@ -140,21 +132,18 @@ func TestPeriodicalTriggerKeepsGoingOnPanic(t *testing.T) {
 	test.WithConcurrencyHarness(t, func(parent context.Context, harness *test.ConcurrencyHarness) {
 
 		logger := mockLogger()
-		x := 0
+		var handlerInvocationCount uint32
 		p := synchronization.NewPeriodicalTrigger(context.Background(), "a periodical trigger", time.Millisecond, logger, func() {
-			x++
+			atomic.AddUint32(&handlerInvocationCount, 1)
 			panic("we should not see this other than the logs")
 		}, nil)
 
-		// more than one error means more than one panic means it recovers correctly
-		for i := 0; i < 2; i++ {
-			<-logger.errors
-		}
+		// a second invocation of the handler means the trigger recovered from the first panic
+		require.True(t, test.Eventually(1*time.Second, func() bool {
+			return atomic.LoadUint32(&handlerInvocationCount) >= 2
```

Comment:
@ronnno maybe extract the value of atomic.Load...() to a variable, to eliminate the chance of confusion by getting different values on the 2 consecutive calls.